app/src/main/java/wilsonoh/sagyo/Main.java

```diff
@@ -0,0 +1,60 @@
+package wilsonoh.sagyo;
+
+import java.util.Scanner;
+
+import wilsonoh.sagyo.commands.Command;
+import wilsonoh.sagyo.exceptions.InvalidTaskException;
+import wilsonoh.sagyo.exceptions.SagyoException;
+import wilsonoh.sagyo.parser.CommandParser;
+import wilsonoh.sagyo.storage.Storage;
+import wilsonoh.sagyo.tasklist.TaskList;
+import wilsonoh.sagyo.ui.TextFormatter;
+
+public class Main {
+
+    private static final String[] GREETING =
+        new String[] {"Welcome to sagyo, your task manager!", "What can I do for you?"};
+
+    private static final String PROMPT = "> ";
+
+    private final TaskList tasks = new TaskList();
+
+    /**
+     * Runner function of the class, executes the main loop
+     */
+    public void run() {
+        TextFormatter ui = new TextFormatter(2, 120);
+        Storage storage = new Storage();
+        try {
+            this.tasks.extendFromStorage(storage);
+        } catch (InvalidTaskException e) {
+            System.out.println(e.getMessage());
+        }
+        CommandParser cmdParser = new CommandParser(tasks);
+        boolean isRunning = true;
+        ui.printLinesInfo(GREETING);
+        // try-with-resources to close the scanner automatically, preventing resource leaks
+        try (Scanner sc = new Scanner(System.in)) {
+            // Loop will break on execution of the ByeCommand
```

Comment:
These comments should be removed if you just use it to self-explain, so that your code looks cleaner